crates/sqlite_store/schema/schema_0.sql

```diff
@@ -0,0 +1,65 @@
+-- schema version control
+CREATE TABLE version (version INTEGER) STRICT;
+INSERT INTO version VALUES (1);
+
+-- network is the valid network for all other table data
+CREATE TABLE network
+(
+    name TEXT UNIQUE NOT NULL
+) STRICT;
+
+-- keychain is the json serialized keychain structure as JSONB,
+-- descriptor is the complete descriptor string,
+-- descriptor_id is a sha256::Hash id of the descriptor string w/o the checksum,
+-- last revealed index is a u32
+CREATE TABLE keychain
+(
+    keychain      BLOB PRIMARY KEY NOT NULL,
+    descriptor    TEXT UNIQUE      NOT NULL,
+    descriptor_id BLOB UNIQUE      NOT NULL,
+    last_revealed INTEGER
+) STRICT;
+
+-- hash is block hash hex string,
+-- block height is a u32,
+CREATE TABLE block
+(
+    hash   TEXT PRIMARY KEY NOT NULL,
+    height INTEGER          NOT NULL
+) STRICT;
+
+-- txid is transaction hash hex string (reversed)
+-- whole_tx is a consensus encoded transaction,
+-- last seen is a u64 unix epoch seconds
+CREATE TABLE tx
+(
+    txid      TEXT PRIMARY KEY NOT NULL,
+    whole_tx  BLOB             NOT NULL,
+    last_seen INTEGER
+) STRICT;
+
+-- Outpoint txid hash hex string (reversed)
+-- Outpoint vout
+-- TxOut value as SATs
+-- TxOut script consensus encoded
+CREATE TABLE txout
+(
+    txid   TEXT    NOT NULL,
+    vout   INTEGER NOT NULL,
+    value  INTEGER NOT NULL,
+    script BLOB    NOT NULL,
+    PRIMARY KEY (txid, vout)
+) STRICT;
+
+-- set of block "anchors" and txids
+-- block hash hex string
+-- confirmation height u32
+-- confirmation time unix epoch seconds
+-- txid is transaction hash hex string (reversed)
+CREATE TABLE anchor
+(
+    block_hash          TEXT NOT NULL REFERENCES block (hash),
+    confirmation_height INTEGER,
+    confirmation_time   INTEGER,
+    txid                TEXT NOT NULL REFERENCES tx (txid)
+) STRICT;
```

Comment:
Would this duplicate anchors stored in the db? The entire `(anchor, txid)` should uniquely identify the row (according to my shallow understanding of SQL).

Reply:
Also, how important is it to store `confirmation_height` and `confirmation_time` as separate fields? We could make this more "generic" if we had an anchor-blob field that is a unique identifier and block_hash + txid.

Reply:
If the software is correct we shouldn't get duplicate `anchor` records, but the purpose of the DB constraints are mainly to catch software issues. So I added a `UNIQUE` constraint for the `anchor` table that includes all the `anchor` fields.

As for `anchor.confirmation_height` and `anchor.confirmation_time` columns I'd prefer to keep them separate to make debugging a little easier.  They're both human readable and if one or both aren't used (depending on anchor type) it doesn't really cost anything to leave them as `NULL`.

Reply:
I can also see how making `anchor` a blob would make this DB schema work with any anchor type. But do you expect there to be any more types? Making things too generic makes it harder to reason about.

Reply:
I do expect more `Anchor` types, and I think it's not too much work to make it flexible. One thing I had in mind is SPV proof anchors. However, I expect that it will be a very useful thing for users to save any anchor type.

In terms of debugging persistence backends, the only thing this is required is to make sure an aggregation of persisted changesets (using `Append::append`) is the same as the changeset recovered from the persistence. Or did you mean for something else?

Reply:
Ok I'll try reworking this with `Anchor` as a JSONB blob. After sleeping on it, I think this will probably simplify the code a lot. 

The type of debugging I was thinking about was for devs to be able to run queries on the DB to double check values they see from their app. But with JSONB we can still query sub-fields of the `Anchor` blob which should be fine.

Reply:
@evanlinjin I pushed my changes to make `Anchor` a blob, it did simplify the code. Here's an example query that shows how someone can still query the `Anchor` data (if they know what they're looking for).
```sql
SELECT block_hash, block_height, datetime(json_extract(anchor, '$.confirmation_time'), 'unixepoch') AS confirmation_time
FROM anchor;
```
```csv
0000006504d5a5efe4bdbd86a7052a4dddcd467b35c2ebde3689099746d9f037,154186,2023-08-01 07:23:03
000000f8721a9c4e1d99b833bdf01174b0cfa9cff0f72926ceee5e08074d2c7c,158758,2023-09-01 19:39:34
000000f4289f0a969dee95f3d23951b9b375f97a5f1545dd93e09032070af48b,158770,2023-09-01 20:40:08
000000f4289f0a969dee95f3d23951b9b375f97a5f1545dd93e09032070af48b,158770,2023-09-01 20:40:08
etc..
```